### ISSUE TITLE
[idea] creating snap of wine runtime

### ISSUE THREAD

**Issue by Maryse47** [mmtrt/wine-platform#1]
Did you considered creating standalone wine runtime as snap? This way it won't need to create separate snap for every app you want to run (we're talking about thousands of windows apps which can be run with wine) but install them through this runtime instead. It would work similar to wine installed as generic package in system but with advantages if being snap.

**Comment by mmtrt** [comment]
Yes it can be done but there is the snap confinement issue where some applications and games can cause `wineserver ptrace` which is very problematic likely will crash or hang. For example if `wine-runtime` is `strict confinement` and `leagueoflegends` game does cause `wineserver ptrace` so users can not play that game at all unless `wine-runtime` is on `devmode` which is that developers should not encourage end-users to install snaps with the ‘–devmode’ switch. Devmode snaps cannot be published in the stable channel of the snap store. End users should not normally install snaps in devmode as this undoes any protections set out with confinement and interfaces. Snaps which use devmode confinement cannot be published in the snap store stable.
However there is `classic confinement` which the snap having it needs to be reviewed and accepted by snappy advocacy team.

You can read more about [snap confinement](https://snapcraft.io/blog/demystifying-snap-confinement).

**Comment by Maryse47** [comment]
I'm absolutely not advocating for disabling snap confinement. I run wine confined with custom apparmor profile which denies `ptrace` (`ptrace` is also restricted by yama) and didn't saw anything unusual. If there are some corner cases like leagueoflegends which abuses `ptrace` then they may require its own snap but majority apps should work just fine.

**Comment by brlin-tw** [comment]
I wonder if we can actually publish one with classic confinement...

Reference: [Is it possible to build Wine with Snappy? - snapcraft - snapcraft.io](https://forum.snapcraft.io/t/is-it-possible-to-build-wine-with-snappy/1855)

**Comment by mmtrt** [comment]
Here is my example proposal post on how this will shape but I'm not doing it.

[Link](https://forum.snapcraft.io/t/since-ubuntus-dropping-32-bit-libraries-support-will-wine-steam-lutris-etc-be-snapped-by-the-time-19-04-is-launched/11963/2?u=mmtrt)
